### ISSUE TITLE
Blink probability doesn't go above 0.001

### ISSUE THREAD

**Issue by TrussedBoat** [fodorad/BlinkLinMulT#3]
I have set up both BlinkLinT and BlinkLinMulT. I created a feature vector extractor using both my own implementation and changing the scripts in blinklinmult/preprocess/ from this repository.

After grouping the eye crops and feature vectors and feeding them into the BlinkLinMulTusing the provided pretrained weights, I noticed that the blink probability never goes above 0.001 (the logits don’t exceed -6)

I encountered the same issue with BlinkLinT as well. However, after setting time_window = 1 and passing the eye crops through DenseNet121, I received more sensible outputs.

Is this an issue with the pretrained weights, or could it be something else?

**Comment by rupalsaxena** [comment]
Hello,

I hope you already solved this issue! I was facing same issue with probability under 0.01! 

I received more sensible results after I normalized my input img vector as shown in code below:
```
def normalize_tensor(x_seq, device):
    x_seq = x_seq / 255.0

    # apply ImageNet mean/std
    mean = torch.tensor([0.485, 0.456, 0.406], device=device).view(1,1,3,1,1)
    std  = torch.tensor([0.229, 0.224, 0.225], device=device).view(1,1,3,1,1)
    x_seq = (x_seq - mean) / std
    return x_seq
```
Thank you!

Best,
Rupal

**Comment by fodorad** [comment]
Hi everyone, thank you for your interest!  

I updated the [exordium](https://github.com/fodorad/exordium) and [blinklinmult](https://github.com/fodorad/BlinkLinMulT) packages, which now include a demo usage of the TalkingFace dataset.  

### Main additions and changes
- **Exordium**: updated for macOS support, reorganized dependencies (lighter install if only using face/eye feature extraction).  
- **BlinkLinMulT**: added TalkingFace data, normalization parameters, and two demos (feature extraction + model inference).  
- **TalkingFace dataset integration**: reshared dataset added with annotation file; unzip under `data/db/talkingface` to match expected structure.  
- **talkingface.json**: includes mean and std values for features (both eyes).  
- **Feature extraction demo**: step-by-step visualization via `blinklinmult/demo/talkingface_feature_extraction.py`.  
- **Benchmarking support**: annotations in Tag format with reader (`blinklinmult.preprocess.reader`).  
- **Automatic pipeline**: RetinaFace + TDDFA_V2 + FaceMesh + MediaPipe Iris integrated for more precise landmark and eye patch extraction.  
- **Inference demo**: `blinklinmult/demo/talkingface_inference.py` produces predictions for the first blink sample using DenseNet121, BlinkLinT, and BlinkLinMulT.  
- **Sample predictions**: all three models tested, sequence models better capture dynamics compared to frame-by-frame eye state.  
- **Threshold tuning**: no preset applied in the demo; suggested values vary from 0.5 (clean samples) up to 0.9 (noisy datasets like RN15, RN30).  


> I created a feature vector extractor using both my own implementation and changing the scripts in blinklinmult/preprocess/ from this repository

This is hopefully resolved by the updated exordium package.

> I received more sensible results after I normalized my input img vector as shown in code below

This is hopefully resolved by using the calculated standardization parameters for all high-level and rgb patches (now its 
simply ImageNet params)

I also attached the outputs of the models on the first blink example, please try to reconstruct it via the demo scripts:

<img width="1000" height="500" alt="Image" src="https://github.com/user-attachments/assets/a1d61332-cd7d-4440-a9cf-f640c554c204" />
<img width="1000" height="500" alt="Image" src="https://github.com/user-attachments/assets/57bb2188-c896-4ea5-bf48-2b0cefe507ed" />
<img width="1000" height="500" alt="Image" src="https://github.com/user-attachments/assets/9b476300-2e8f-4370-a13c-e1e7330e1449" />

I hope this will help you get started. Over the coming weeks, I’ll continue refactoring parts of the internal (non-sharable) repository so that I can eventually publish the full functional codebase with training support, not just inference.  

Please pull the latest updates and take a look. Let me know if you have any questions—I’m available for further discussions!
I will close this issue if the initial problem is solved. Any further, more specific requests may have their own separate threads—Cheers!

Best,
Adam
